From 5d4b0dad7066f318079e33161d14ca207c7050ca Mon Sep 17 00:00:00 2001
From: Paul Kocialkowski <contact@paulk.fr>
Date: Sat, 22 Apr 2017 13:33:22 +0200
Subject: [PATCH 1/2] ARM: tegra: jetson-tk1: Use EHCI instead of XHCI

While XHCI (USB3) requires a loaded, proprietary firmware, EHCI (USB2)
is a hardware-only implementation. Switching to EHCI by default allows
shipping the kernel without the proprietary XHCI firmware, making USB
work out of the box in free system distributions.

Signed-off-by: Paul Kocialkowski <contact@paulk.fr>
---
 arch/arm/boot/dts/tegra124-jetson-tk1.dts | 38 ++++++-------------------------
 1 file changed, 7 insertions(+), 31 deletions(-)

diff --git a/arch/arm/boot/dts/tegra124-jetson-tk1.dts b/arch/arm/boot/dts/tegra124-jetson-tk1.dts
index e8e777b8ef1b..0bcaf989720c 100644
--- a/arch/arm/boot/dts/tegra124-jetson-tk1.dts
+++ b/arch/arm/boot/dts/tegra124-jetson-tk1.dts
@@ -1695,25 +1695,6 @@
 		status = "okay";
 	};
 
-	usb@70090000 {
-		phys = <&{/padctl@7009f000/pads/usb2/lanes/usb2-0}>, /* Micro A/B */
-		       <&{/padctl@7009f000/pads/usb2/lanes/usb2-1}>, /* Mini PCIe */
-		       <&{/padctl@7009f000/pads/usb2/lanes/usb2-2}>, /* USB3 */
-		       <&{/padctl@7009f000/pads/pcie/lanes/pcie-0}>; /* USB3 */
-		phy-names = "usb2-0", "usb2-1", "usb2-2", "usb3-0";
-
-		avddio-pex-supply = <&vdd_1v05_run>;
-		dvddio-pex-supply = <&vdd_1v05_run>;
-		avdd-usb-supply = <&vdd_3v3_lp0>;
-		avdd-pll-utmip-supply = <&vddio_1v8>;
-		avdd-pll-erefe-supply = <&avdd_1v05_run>;
-		avdd-usb-ss-pll-supply = <&vdd_1v05_run>;
-		hvdd-usb-ss-supply = <&vdd_3v3_lp0>;
-		hvdd-usb-ss-pll-e-supply = <&vdd_3v3_lp0>;
-
-		status = "okay";
-	};
-
 	padctl@7009f000 {
 		status = "okay";
 
@@ -1728,12 +1709,12 @@
 					};
 
 					usb2-1 {
-						nvidia,function = "xusb";
+						nvidia,function = "snps";
 						status = "okay";
 					};
 
 					usb2-2 {
-						nvidia,function = "xusb";
+						nvidia,function = "snps";
 						status = "okay";
 					};
 				};
@@ -1743,11 +1724,6 @@
 				status = "okay";
 
 				lanes {
-					pcie-0 {
-						nvidia,function = "usb3-ss";
-						status = "okay";
-					};
-
 					pcie-2 {
 						nvidia,function = "pcie";
 						status = "okay";
@@ -1792,11 +1768,6 @@
 
 				vbus-supply = <&vdd_usb3_vbus>;
 			};
-
-			usb3-0 {
-				nvidia,usb2-companion = <2>;
-				status = "okay";
-			};
 		};
 	};
 
@@ -1840,6 +1811,11 @@
 		status = "okay";
 	};
 
+	usb-phy@7d000000 {
+		status = "okay";
+		vbus-supply = <&vdd_usb1_vbus>;
+	};
+
 	/* mini-PCIe USB */
 	usb@7d004000 {
 		status = "okay";
-- 
2.15.0

